### PR TITLE
devops: push Docker image to MCR

### LINKED ISSUE #290: Publish docker image to MCR
This would enable users to pull the image for their CI environments.

### PR DESCRIPTION
**Pull request by mxschmitt**
Closes #290

`DOCKER_PASSWORD` environment variable needs to be set.

### CODE REVIEW

**Comment by mxschmitt** [comment]
I don't get it why its failing, maybe someone has an idea?

**Comment by mxschmitt** [comment]
@pavelfeldman not sure why its failing, only the video test, hmm.

**Comment by kumaraditya303** [comment]
> I don't get it why its failing, maybe someone has an idea?

@mxschmitt It is failing because of timeout, remove timeout and all the tests would pass.

File: .github/workflows/publish_canary_docker.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: "devrelease:docker"
```
Comment (pavelfeldman) [inline .github/workflows/publish_canary_docker.yml:1]: Curious, why do we need this one?

Comment (mxschmitt) [reply .github/workflows/publish_canary_docker.yml:1]: We can also remove the canary builds. They were useful for testing and Lushnikiv and me used them a lot so I think it brings value.

**Review by aslushnikov — APPROVED** [review]

**Review by aslushnikov — CHANGES_REQUESTED** [review]
Marking as "request changes" to not merge accidentally since this the tests are actually failing in the container.

@mxschmitt is looking into this

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**